src/list/index.tsx

```diff
@@ -726,17 +726,16 @@ export const List = factory(function List({
 	let scrollTop = icache.getOrSet('scrollTop', 0);
 	const nodePadding = Math.min(calculatedItemsInView, 20);
 	const renderedItemsCount = calculatedItemsInView + 2 * nodePadding;
-	let computedActiveIndex =
-		activeIndex === undefined ? icache.getOrSet('activeIndex', 0) : activeIndex;
+	let computedActiveIndex = activeIndex === undefined ? icache.get('activeIndex') : activeIndex;
 	const inputText = icache.get('inputText');
 	const {
 		meta: { total = 0 }
 	} = get(options(), { meta: true, read });
 	if (inputText && inputText !== icache.get('previousInputText') && total) {
 		const items = get({ ...options(), offset: 0, size: total });
-		const first = items.slice(0, computedActiveIndex);
-		const second = items.slice(computedActiveIndex);
-		let foundIndex = computedActiveIndex;
+		const first = computedActiveIndex ? items.slice(0, computedActiveIndex) : [];
```

Comment:
if computed active index was `0` wouldn't this be falsey rather than selecting the first item?